functions-continued/variables-are-references.md

```diff
@@ -0,0 +1,648 @@
+==== Adding 777 to score ====
+id of score: 4551559440
+==== After calling add_lucky_bonus ====
+id of jackfruit: 4539856336
+value of jackfruit: 222
+```
+
+We should observe:
+
+1. `jackfruit` points to an immutable object
+1. When we call `add_lucky_bonus`
+   - We pass in `jackfruit`
+   - `score` initially points to the same object as `jackfruit`
+   - After `score += 777`, `score` points to a different immutable object
+1. `jackfruit` continues to point to the same immutable object
+
+## Extending the Concept: Nested Data Structures
+
+Lists and dictionaries are not only mutable data types, but they're also _containers_. As containers, they contain either mutable or immutable data types itself!
+
+The same principles extend to nested data structures.
+
+Consider the following questions:
+
+- Imagine a list within a list. If we modify the inner list, does that affect the references pointing to it?
+- Imagine a dictionary within a list. What are the effects of modifying the inner dictionary?
+
+Make a prediction. Run an experiment.
+
+<!-- Question  -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: paragraph
+* id: tRHUpg
+* title: Variables Are References
+##### !question
+
+Predict:
+
+- Imagine a list within a list. If we modify the inner list, does that affect the references pointing to it?
+- Imagine a dictionary within a list. What are the effects of modifying the inner dictionary?
+
+##### !end-question
+##### !placeholder
+
+I predict...
+
+##### !end-placeholder
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+Now, read through this code and its example output:
+
+```python
+peaches = ["Hello", [False, False, False] ,"World!"]
+
+pears = peaches[1]
+
+pears.append(True)
+
+print("value of peaches:", peaches)
+```
+
+```
+value of peaches: ['Hello', [False, False, False, True], 'World!']
+```
+
+`peaches[1]` _and_ `pears` both point to the mutable object `[False, False, False]`.
+
+When we modify that object with `pears.append()`, we modify the same object referred to by `peaches[1]`.
+
+Let's look at an example dictionary inside a list:
+
+```python
+pears = {
+    "hello": "world!"
+}
+
+peaches = [False, pears, False]
+
+pears["hello"] = "modified value"
+
+print("value of peaches:", peaches)
+```
+
+```
+value of peaches: [False, {'hello': 'modified value'}, False]
+```
+
+`pears` and `peaches[1]` refer to the same dictionary object.
+
+Giving a new value to a key in a dictionary counts as modification. Modifying `pear` affects the item at `peaches[1]`.
+
+### !callout-info
+
+## Endless Combinations
+
+Curious about more combinations, such as a list in a dictionary, or a list inside a list inside a list? Follow your curiosity and experiment!
+
+### !end-callout
+
+### !callout-success
+
+## Foreshadowing: OOP
+
+When discussing object-oriented programming, we'll write user-defined classes. User-defined classes are mutable data types by default.
+
+### !end-callout
+
+## Best Pattern: Reassign Often, Return Often
```

Comment:
How about this instead? I think it emphasizes the importance of using return values, which seems like the most important take away from this.
```suggestion
## Prefer Returning Results Over Modifying Inputs
```